### DISCUSSION TITLE
Sitemap.xml

### DISCUSSION THREAD

**Discussion by CarolinePublipresse** [OFFLINE-GmbH/oc-boxes-support#55] [Ideas and feature requests]
I didn't found on your documentation that Boxes provided integration with Rainlab.Sitemap. But I found it on your classes ^^ 

It might seem obvious for projects where you need Rainlab.Pages that it will be compatible with the sitemap module, but I didn't need Rainlab.Pages, so I was preparing to make a sitemap.xml from scrach. 

I didn't had to, thanks !
Rainlab.Sitemap works perfectly to provide all my Box pages in the sitemap.xml. This should appear in your docs ;)

**Comment by tobias-kuendig** [comment]
Hi @CarolinePublipresse 

Thank you for the feedback. We do have integration for the Sitemap plugin, that is correct. Boxes also integrates great with Rainlab.Pages or with the new `| link` filter in October 3.0.

Since the October devs kept the API for all these use-cases the same, it was easy enough for us to implement all the features.

You are right that this needs to go into the docs. I have created a reminder on our internal tracker to add this with a future release.

**Reply by chrisvidal** [reply]
Hi @tobias-kuendig 
wanted to check with regards to the sitemap integration behavior as i don't find it in the doc.

It seems to work as the sitemap plugin is offering up to choose `Web page` or `All Web Pages`. 

However, I am not sure how the multisite aspect is taken in account ?

the current result in my `sitemap.xml` looks like:

```xml
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"...>
...
  <url>
    <loc>http://domain.test/about</loc>
    <lastmod>2025-04-01T13:47:58+00:00</lastmod>
    <changefreq>weekly</changefreq>
    <priority>0.5</priority>
  </url>
...
</urlset>
```

which does not take into account the 2 sites definitions used in my multisite config.

Is that expected ? Is that more of a problem with the sitemap plugin maybe ?

thanks in advance

**Reply by chrisvidal** [reply]
it looks like the `sitemap` plugin is expecting a `alternate_locale_urls` attribute on the items coming from the `resolveMenuItems` external function.

**Comment by tobias-kuendig** [comment]
v3.6.68 generates the alternate_locale_urls keys, let me know if this is enough.

**Reply by chrisvidal** [reply]
thank you @tobias-kuendig.

I am not sure this is sufficient as it generates only one language when I have 2 sites. Probably a limitation from the `sitemap` plugin

```xml
<url>
<loc>http://domain.test/about</loc>
<lastmod>2025-04-02T07:36:40+00:00</lastmod>
<changefreq>weekly</changefreq>
<priority>0.5</priority>
<xhtml:link rel="alternate" hreflang="en" href="http://domain.test/en/about"/>
</url>
```

**Reply by tobias-kuendig** [reply]
Please try v3.6.69, I was able to fix another issue.
